### ISSUE TITLE
[Datepicker-Simple]: Require an option to remove year from the calendar.

### ISSUE THREAD

**Issue by Monika-Jayanna** [carbon-design-system/carbon#11556]
### The problem

I am using the following design in our application
https://react.carbondesignsystem.com/?path=/docs/components-datepicker--simple
and its working fine but as per our business requirement we need to remove the year option we tried with changing the date format option but it still requires an user to give Year option, So we need an option to remove the year from calendar.

<img width="808" alt="image" src="https://user-images.githubusercontent.com/97293722/172152891-8e71d337-5775-43e3-9361-5554966acc3e.png">

### The solution

So it would be very helpful if we get an option to completely **_remove year month and date_** in date picker component, based on our requirements from the user perspective interface.

### Application/website

https://www.ibm.com/privacy/portal/us-en

### Business priority

High Priority = pressing release

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by abbeyhrt** [comment]
Hey @Monika-Jayanna! It looks like you can update the `dateFormat` and `placeholder` to remove the year from the DatePicker. The year still shows up in the dropdown calendar but it defaults to the current year. Here is a code example to show what I mean: https://stackblitz.com/edit/github-j7hrws?file=src/App.jsx does that satisfy the business requirements you are looking for?

**Comment by Monika-Jayanna** [comment]
@abbeyhrt Yes we have tried this but we do not want to show the _**current year**_ also it should not display the year in anywhere of the calendar and we are not using _**date ranges**_ in our application, We are just using the simple calendar with only one input option of date picker component.
And one more thing we have observed is when we have removed the **_year_** option from **_date format_** it was giving an **_validation error_** to match with the format even though we were not using the validation option.

**Comment by Monika-Jayanna** [comment]
Hi Team, Could you please give the status/Update of this ticket

**Comment by mbgower** [comment]
You could also achieve this by restricting the date to a specific year using min/max, as in this example from MDN
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

It's still going to display the year, but it will show as disabled and be non-interactive. IMO, this is a more prudent thing to do than hiding the year, which could cause unanticipated issues for users. There may be edge cases even within your use case where a user _does_ need an ability to review a stored date or traverse the period between December and January. 
<img width="764" alt="image" src="https://user-images.githubusercontent.com/14298245/178995373-77033bf7-b699-43cf-bbc4-b6092abb20ba.png">
Note that I haven't seen what setting a range of only one year does with Carbon. I'm just citing a default html date input
